service/types.go

```diff
@@ -0,0 +1,21 @@
+package service
+
+import "github.com/spacemeshos/poet/shared"
+
+type Proof struct {
```

Comment:
If these types are not intended to be used outside of the `service` package, lets make them private.